### PR TITLE
Beam as structure element

### PR DESCRIPTION
**Pull request by e-koch**
Implements an ```EllipticalTophat2DKernel```, allowing for the beam to be converted into a structure element for morphological operations. This required implementing an ```EllipticalDisk2D``` model as well.

### CODE REVIEW

File: radio_beam/beam.py [hunk 1]
```diff
@@ -522,3 +535,84 @@ def __init__(self, width, height, position_angle, support_scaling=8, **kwargs):
...
+    theta : float
+        Orientation of the major axis with respect to the x axis.
+
+    See Also
+    --------
+    Box2D, TrapezoidDisk2D, Disk2D
+
+    Notes
+    -----
+    ** NOT UPDATED **
```
Comment (keflavich) [inline radio_beam/beam.py]: not updated from what?

File: radio_beam/beam.py [hunk 2]
```diff
@@ -429,6 +435,16 @@ def as_kernel(self, pixscale):
                                           self.minor.to(u.deg).value/pixscale,
                                           self.pa.to(u.radian).value)
 
+    def as_struct_element(self, pixscale):
```
Comment (keflavich) [inline radio_beam/beam.py]: This needs a docstring.  I'm not sure about the name, either - why not `as_elliptical_kernel`, for example?

File: radio_beam/beam.py (hunk 1 above)
Comment (e-koch) [inline radio_beam/beam.py]: Ah. Forgot to change that. I copied ```Disk2D``` from astropy and haven't updated to the elliptical model.

File: radio_beam/beam.py (hunk 2 above)
Comment (e-koch) [inline radio_beam/beam.py]: Does that distinguish enough from ```as_kernel```? Maybe something like ```as_boolean_kernel```?

Comment (keflavich) [inline radio_beam/beam.py]: AH!  I had missed that this is an `EllipticalDisk`, not an `EllipticalGaussian`.  Yes, maybe `as_boolean_kernel` or `as_elliptical_tophat_kernel`?  Or just `as_tophat_kernel`?  Is there a type of top hat that is elliptical?

Comment (e-koch) [inline radio_beam/beam.py]: You mean in general an elliptical top hat? I'm not too sure... that just seemed like the obvious name for it.

I like ```as_tophat_kernel```.

Comment (keflavich) [inline radio_beam/beam.py]: OK, go with it (`as_tophat_kernel`).  However, I think in the present formulation, the area is not the same as the Gaussian kernel, and it should be.  The volume of a 2D gaussian is `2 pi sigma^2`, or `2*np.pi*major*minor/2.35**2`, so the effective major/minor axes of the tophat kernel should be `r_maj = major/sqrt(4*log(2))` if I'm doing my math right.

Comment (e-koch) [inline radio_beam/beam.py]: ```sqrt(4*log(2))``` or ```sqrt(8*log(2))```?

Comment (keflavich) [inline radio_beam/beam.py]: I did a shortcut here: I think it's `sqrt(4*log(2))`
for circular, using mixed notation:
`A_gaussian = 2 * pi * sigma^2 / (sqrt(8*log(2))^2`
`A_tophat = pi * r^2`
`pi r^2 = 2 * pi * sigma^2 / (sqrt(8*log(2))^2`
`r = sqrt(2)/sqrt(8*log(2)) * sigma`

Comment (e-koch) [inline radio_beam/beam.py]: Ah ok, that makes sense. Thanks!

**Comment by e-koch** [comment]
@keflavich - The tophat area should now match the gaussian. I'm going to do a bit more testing tomorrow though.

**Comment by keflavich** [comment]
@e-koch ok just ping me when you want it merged

**Comment by e-koch** [comment]
@keflavich - Good to merge!

**Comment by e-koch** [comment]
at0206_beam_kernels

**Comment by e-koch** [comment]
Example of a beam and the elliptical gaussian (image) with the elliptical tophat (contour).

**Comment by keflavich** [comment]
Octagonal ellipses are my favorite ellipses.

**Pull request merged**